apps/src/gamelab/spritelab/actionCommands.js

```diff
@@ -0,0 +1,150 @@
+import * as spriteUtils from './spriteUtils';
+import {commands as behaviorCommands} from './behaviorCommands';
+
+export const commands = {
+  changePropBy(spriteId, prop, val) {
+    if (!val) {
+      return;
+    }
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    let specialCases = {
+      direction: sprite => (sprite.direction = (sprite.direction + val) % 360),
+      scale: sprite => {
+        sprite.setScale(sprite.getScale() + val / 100);
+        if (sprite.scale < 0) {
+          sprite.scale = 0;
+        }
+      },
+      y: sprite => (sprite.y -= val)
+    };
+    sprites.forEach(sprite => {
+      if (specialCases[prop]) {
+        specialCases[prop](sprite);
+      } else {
+        sprite[prop] += val;
+      }
+    });
+  },
+  edgesDisplace(spriteId) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    sprites.forEach(sprite => this.edges.displace(sprite));
+  },
+  isTouchingEdges(spriteId) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    let touching = false;
+    sprites.forEach(sprite => {
+      if (sprite.isTouching(this.edges)) {
+        touching = true;
+      }
+    });
+    return touching;
+  },
+  jumpTo(spriteId, location) {
+    if (!location) {
+      return;
+    }
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    sprites.forEach(sprite => {
+      sprite.x = location.x;
+      sprite.y = location.y;
+    });
+  },
+  mirrorSprite(spriteId, direction) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    sprites.forEach(sprite => {
+      if (direction === 'right') {
+        sprite.mirrorX(1);
+      } else {
+        sprite.mirrorX(-1);
+      }
+    });
+  },
+  moveForward(spriteId, distance) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    sprites.forEach(sprite => {
+      if (!sprite.direction) {
+        sprite.direction = 0;
+      }
+      let direction = sprite.direction % 360;
+      sprite.x += distance * Math.cos((direction * Math.PI) / 180);
+      sprite.y += distance * Math.sin((direction * Math.PI) / 180);
+    });
+  },
+  moveInDirection(spriteId, distance, direction) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    let dirs = {
+      North: sprite => (sprite.y -= distance),
+      East: sprite => (sprite.x += distance),
+      South: sprite => (sprite.y += distance),
+      West: sprite => (sprite.x -= distance)
+    };
+    if (!dirs[direction]) {
+      console.error('invalid direction: ' + direction);
+    }
+    sprites.forEach(sprite => {
+      dirs[direction](sprite);
+    });
+  },
+  moveToward(spriteId, distance, target) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    sprites.forEach(sprite => {
+      if (sprite && target) {
+        let angle = Math.atan2(target.y - sprite.y, target.x - sprite.x);
+        if (angle) {
```

Comment:
Angle could be 0 when there is still distance between the two which would cause this to evaluate to false. i.e. 
`Math.atan2(5 - 5, 50 - 0)`

Reply:
👍